weave

```diff
@@ -805,6 +811,18 @@ proxy_args() {
     done
 }
 
+proxy_addr() {
+    if addr=$(docker logs weaveproxy | grep -oE "proxy listening on .*"); then
```

Comment:
I wonder how this will perform when the proxy log is long. Is there a way to short-circuit this? Would adding a reasonably-sized `head` work? Or would the sub-expression still only terminate after the entire log has been retrieved?